src/InputNumber.tsx

```diff
@@ -380,10 +389,16 @@ const InternalInputNumber = React.forwardRef(
 
     // >>> Collect input value
     const collectInputValue = (inputStr: string) => {
+      // validate string
+      if (validator) {
+        if (!validator(inputStr)) return;
+      }
```

Comment:
InputNumber does not currently use error state or messages to show users, the idea is just to prevent the value from being updated if the validation fails.